src/main/java/net/openhft/chronicle/core/OS.java

```diff
@@ -115,6 +115,19 @@
     private OS() {
     }
 
+    private static String findTmp() {
+        String target = System.getProperty("project.build.directory");
+        if (target != null)
+            return target + File.pathSeparator + "tmp";
+        final String tmp = System.getProperty("java.io.tmpdir");
+        if (tmp != null
+                && new File(tmp).isDirectory()
+                && new File(tmp).canWrite())
+            return tmp;
+        new File("tmp").mkdirs();
```

Comment:
what if mkdirs() fails?

Reply:
A subsequent IO operation that needs it will fail.

Reply:
sonar will warn about ignoring the return value - maybe we should at the very least assert the return value?